lua/fff/rust/path_utils.rs

```diff
@@ -1,90 +1,333 @@
-pub fn calculate_distance_penalty(current_file: Option<&str>, candidate_path: &str) -> i32 {
-    let Some(ref current_path) = current_file else {
-        return 0; // No penalty if no current file
+const MAX_PENALTY_LEVEL_MULTIPLIER: i32 = 10;
+
+pub fn calculate_filename_similarity_bonus(
+    current_file_path: &str,
+    candidate_file_path: &str,
+    max_bonus: i32,
+    similarity_threshold: f64,
+) -> i32 {
+    use std::path::Path;
+    use strsim::jaro_winkler;
```

Comment:
I think Jaro Winkler is a good choice since:
1. Prefix bias: Jaro-Winkler gives higher scores to strings that share common prefixes, which aligns with how developers typically name related files (e.g., user.rs, user_test.rs, user_service.rs)
  2. Typo tolerance: Handles minor typos and variations gracefully, which is important for fuzzy file finding
  3. Performance: It's computationally efficient compared to algorithms like Levenshtein.
  4. Score range: Returns normalized scores between 0.0 and 1.0, making it easy to set meaningful thresholds